src/common/lib/hid_interface.c

```
@@ -33,124 +33,149 @@
-    char buffer[128];
-    const char* prefix = result ? "[SENT-HID-REPORT]" : "[FAILED-HID-REPORT]";
-    size_t offset = (size_t)snprintf(buffer, sizeof(buffer), "%s %02X ", prefix, report_id);
-    
-    const uint8_t* p = (const uint8_t*)report;
-    for (uint16_t i = 0; i < len && offset < sizeof(buffer) - 4; i++) {
-      offset += (size_t)snprintf(buffer + offset, sizeof(buffer) - offset, "%02X ", *p++);
+static inline bool hid_send_report(uint8_t instance, uint8_t report_id, void const* report,
+                                   uint16_t len) {
+    // Check if endpoint is ready before attempting to send
+    bool ready = tud_hid_n_ready(instance);
+    if (!ready) {
```

Clang (14.0.6)

[note] +1, including nesting penalty of 0, nesting level increased to 1

(clang)